Updating to 1.15.2 and build simplification

paulolc opened the pull request:
Compilation now supports 1.15.2 (should resolve #517)
Build on top of @pawel8273 changes for 1.14.x.

mizpablito committed:
Updating to 1.14.x

paulolc committed:
Updating to 1.15.x

paulolc committed:
Build simplified. Now "mvn install" builds everything, including spigot if necessary

paulolc commented:
- Build simplified. Now "mvn install" builds everything, including spigot if necessary.
- Spigot BuiltTools is only run if spigot jar is not detected in local maven repository 
- Restructured pom structure with a parent pom and 2 modules "spigot" and "jsonapi"
- Removed build scripts now obsolete
- Removed all warning from build (except the Vault one, which is a warning to an external pom)
- Version is no longer taken from maven parameters. It's now fixed on pom.xml
- Dropped support for building to older versions of MC. Now only builds for the latest as of now: 1.15.2

alecgorge commented on jsonapi/src/main/java/com/alecgorge/minecraft/jsonapi/packets/netty/NettyInjector.java:
```diff
@@ -79,9 +80,10 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
-            networkManagers = (List<Object>) FuzzyReflection.fromObject(serverConnection, true).
-                invokeMethod(null, "getNetworkManagers", List.class, serverConnection);
-            
+            //networkManagers = (List<Object>) FuzzyReflection.fromObject(serverConnection, true).
+            //    invokeMethod(null, "getNetworkManagers", List.class, serverConnection);
```
why was this commented out?

my recollection was that this needs to be captured so that we don't interfere with the processing of packets for normal gameplay.

paulolc replied:
Nice catch! I forgot that I had commented that out. I need to check it. Thanks!

It was raising a runtime exception due to not recognizing the getNetworkManagers method signature (maybe something to do with the 1.15.2 nms) and the method would not go until the end and jsonapi would break somewhere else. I commented it out just to see how far it went and it would run ok. Meanwhile I went full throttle on the build simplification and completely forgot about this. I'll have to recheck it.

paulolc committed:
Reintroduced getNetworkManagers in NettyInjector
Removed dead code
Replaced debug preprocessing flag with configurable option
Turned the forward ports checking verification on startup is now configurable (disabled by default).
